packages/circuit-ui/components/ListItem/ListItem.tsx

```diff
@@ -0,0 +1,295 @@
+/**
+ * Copyright 2021, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ReactNode, forwardRef, Ref, HTMLProps, FC } from 'react';
+import { css } from '@emotion/core';
+import isPropValid from '@emotion/is-prop-valid';
+import { ChevronRight, IconProps } from '@sumup/icons';
+import { Dispatch as TrackingProps } from '@sumup/collector';
+
+import styled, { StyleProps } from '../../styles/styled';
+import { disableVisually, focusVisible } from '../../styles/style-mixins';
+import { ReturnType } from '../../types/return-type';
+import { ClickEvent } from '../../types/events';
+import { isFunction } from '../../util/type-check';
+import { useComponents } from '../ComponentsContext';
+import { useClickEvent } from '../../hooks/useClickEvent';
+
+type Variant = 'action' | 'navigation';
+
+interface BaseProps {
+  /**
+   * Choose between 'action' and 'navigation' variant.
+   * Default: 'navigation' if 'href' prop is present, otherwise 'action'.
+   */
+  variant?: Variant;
+  /**
+   * Display a leading icon or status image in addition to the text to help to identify the type or status.
+   */
+  icon?: FC<IconProps> | ReactNode;
+  /**
+   * Display a main label for this list item.
+   */
+  label: ReactNode;
+  /**
+   * Display a status line for this list item.
+   */
+  status?: ReactNode;
+  /**
+   * Display a trailing label or action for this list item.
+   * An additional chevron icon will be rendered when using the 'navigation' variant.
+   */
+  suffix?: ReactNode;
+  /**
+   * Visually mark the list item as selected.
+   */
+  selected?: boolean;
+  /**
+   * Visually and functionally disable the list item.
+   */
+  disabled?: boolean;
+  /**
+   * Visually highlight the list item.
+   */
+  highlighted?: boolean;
```

Comment:
What was the use case for this variant again? If it was for the notification inbox, I think we decided we could leave it out for now (also there will be many more a11y implications for this).

Reply:
Yeah, we talked about leaving this out. I'll scrap it. We can always add it at a later stage when we resolve the a11y issues.